src/impl_/pyclass.rs

```diff
@@ -1013,85 +1013,64 @@ pub trait PyClassThreadChecker<T>: Sized {
     private_decl! {}
 }
 
-/// Stub checker for `Send` types.
+/// Default thread checker for `#[pyclass]`.
+///
+/// Keeping the T: Send bound here slightly improves the compile
```

Comment:
But does that also imply keeping the generic parameter on the struct itself?

Reply:
Yes, here's the error message with `SendablePyClass<T: Send>`:

```
error[E0277]: `Rc<i32>` cannot be sent between threads safely
 --> tests/ui/pyclass_send.rs:4:1
  |
4 | #[pyclass]
  | ^^^^^^^^^^ `Rc<i32>` cannot be sent between threads safely
  |
  = help: within `NotThreadSafe`, the trait `Send` is not implemented for `Rc<i32>`
  = note: use `std::sync::Arc` instead of `std::rc::Rc`
note: required because it appears within the type `NotThreadSafe`
 --> tests/ui/pyclass_send.rs:5:8
  |
5 | struct NotThreadSafe {
  |        ^^^^^^^^^^^^^
note: required by a bound in `SendablePyClass`
 --> src/impl_/pyclass.rs
  |
  | pub struct SendablePyClass<T: Send>(PhantomData<T>);
  |                               ^^^^ required by this bound in `SendablePyClass`
  = note: this error originates in the attribute macro `pyclass` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Here's the error message once the generic parameter is removed:

```
error[E0277]: `Rc<i32>` cannot be sent between threads safely
 --> tests/ui/pyclass_send.rs:4:1
  |
4 | #[pyclass]
  | ^^^^^^^^^^ `Rc<i32>` cannot be sent between threads safely
  |
  = help: within `NotThreadSafe`, the trait `Send` is not implemented for `Rc<i32>`
  = help: the trait `pyo3::impl_::pyclass::PyClassThreadChecker<T>` is implemented for `SendablePyClass`
note: required because it appears within the type `NotThreadSafe`
 --> tests/ui/pyclass_send.rs:5:8
  |
5 | struct NotThreadSafe {
  |        ^^^^^^^^^^^^^
  = note: required for `SendablePyClass` to implement `pyo3::impl_::pyclass::PyClassThreadChecker<NotThreadSafe>`
note: required by a bound in `PyClassImpl::ThreadChecker`
 --> src/impl_/pyclass.rs
  |
  |     type ThreadChecker: PyClassThreadChecker<Self>;
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `PyClassImpl::ThreadChecker`
  = note: this error originates in the attribute macro `pyclass` (in Nightly builds, run with -Z macro-backtrace for more info)
```

There's not a huge amount in it. The one with the generic parameter gets `Send` requirement directly underlined, after removing the generic parameter the error spills onto the trait implementation and doesn't give such a clear hint that `Send` should be implemented IMO.

So I'm going to keep the generic parameter for now and we can decide whether to remove it another time.

Reply:
Fair enough.